pkg/relay/ethereum/events.go

```diff
@@ -62,21 +67,43 @@ func (chain *Chain) findSentPackets(ctx core.QueryContext, fromHeight uint64) (c
 		dstChannelID = channel.Counterparty.ChannelId
 	}
 
-	query := ethereum.FilterQuery{
-		FromBlock: big.NewInt(int64(fromHeight)),
-		ToBlock:   big.NewInt(int64(ctx.Height().GetRevisionHeight())),
-		Addresses: []common.Address{
-			chain.config.IBCAddress(),
-		},
-		Topics: [][]common.Hash{{
-			abiSendPacket.ID,
-		}},
+	blocksPerQuery := chain.config.BlocksPerQuery
+	if blocksPerQuery == 0 {
+		blocksPerQuery = BlocksPerQueryDefault
 	}
 
-	logs, err := chain.client.FilterLogs(ctx.Context(), query)
-	if err != nil {
-		logger.Error("failed to filter logs", err)
-		return nil, err
+	toHeight := ctx.Height().GetRevisionHeight()
+	totalBlocks := toHeight - fromHeight
+	loopCount := totalBlocks / blocksPerQuery
+	if totalBlocks%blocksPerQuery != 0 {
+		loopCount++
+	}
+	var logs []types.Log
+	for i := uint64(0); i < loopCount; i++ {
+		var endBlockNum uint64
+		if i == loopCount-1 {
+			endBlockNum = toHeight
+		} else {
+			endBlockNum = fromHeight + (i+1)*blocksPerQuery - 1
+		}
```

Comment:
This logic is incorrect. `totalBlocks` equals to  `toHeight - fromHeight + 1`.

Reply:
Fix loop count logic
https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/36/commits/2b803bc9d05170cb124cfb38f1f8971fde0bb721